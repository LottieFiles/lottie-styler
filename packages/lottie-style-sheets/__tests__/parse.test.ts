/**
 * Copyright 2023 Design Barn Inc.
 */

import { parse } from '..';

import styles from './__fixtures__/styles.lss';

test('parse', () => {
  expect(parse(styles)).toMatchSnapshot();
});
