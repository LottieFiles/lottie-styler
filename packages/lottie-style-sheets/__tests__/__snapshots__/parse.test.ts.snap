// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "important": undefined,
          "property": "fill-color",
          "type": "declaration",
          "value": "red",
        },
        Object {
          "important": undefined,
          "property": "opacity",
          "type": "declaration",
          "value": "0.2",
        },
      ],
      "selectors": Array [
        Object {
          "type": "id-selector",
          "value": "id1",
        },
        Object {
          "type": "id-selector",
          "value": "id2",
        },
        Object {
          "type": "id-selector",
          "value": "id3",
        },
      ],
      "type": "rule",
    },
    Object {
      "children": Array [
        Object {
          "important": undefined,
          "property": "stoke-color",
          "type": "declaration",
          "value": "#fff",
        },
      ],
      "selectors": Array [
        Object {
          "type": "class-selector",
          "value": "class1",
        },
        Object {
          "type": "class-selector",
          "value": "class2",
        },
        Object {
          "type": "class-selector",
          "value": "class3",
        },
      ],
      "type": "rule",
    },
    Object {
      "children": Array [
        Object {
          "important": undefined,
          "property": "fill-rule",
          "type": "declaration",
          "value": "solid",
        },
      ],
      "selectors": Array [
        Object {
          "type": "class-selector",
          "value": "class",
        },
      ],
      "type": "rule",
    },
  ],
  "type": "root",
}
`;
